src/navigation.ts

```diff
@@ -1356,45 +1356,27 @@ export class Navigation {
    * @returns True if the paste was sucessful, false otherwise.
    */
   paste(copyData: Blockly.ICopyData, workspace: Blockly.WorkspaceSvg): boolean {
-    let isHandled = false;
+    // Do this before clipoard.paste due to cursor/focus workaround in getCurNode.
+    const targetNode = workspace.getCursor()?.getCurNode();
+
     Blockly.Events.setGroup(true);
     const block = Blockly.clipboard.paste(
       copyData,
       workspace,
     ) as Blockly.BlockSvg;
     if (block) {
-      isHandled = this.insertPastedBlock(workspace, block);
```

Comment:
I think this isHandled bit was wrong as it returns false when insertPastedBlock didn't manage to connect the blocks. But if Blockly.clipboard.paste succeeded then we did handle the action at least by putting the block on the workspace.

Reply:
Agree since we don't currently do anything different if the block went onto the workspace rather than being connectable; we can always add logic back later if we want to do anything extra to handle that case.